File: NPB-CPP-cadna/cadna_c-3.1.11/examplesC/ex4.c
#include <math.h>
#include <stdio.h>

int main()
{
  int i;
  double a,b,c;

  printf("-------------------------------------\n");
  printf("| A second order recurrent sequence |\n");
  printf("| without CADNA                     |\n");
  printf("-------------------------------------\n");

  a = 5.5;
  b = 61./11.;
  for(i=3;i<=30;i++){
    c = b;
    b = 111. - 1130./b + 3000./(a*b);
    a = c;
    printf("U(%d) = %+.15e\n",i,b);
  }
  printf("The exact limit is 6.\n");

  return 0;
}

File: NPB-CPP-cadna/cadna_c-3.1.11/examplesFortran/ex4.f90
      program ex4
      implicit double precision (a-h,o-z)
      !
      print *,'A second order recurrent sequence without CADNA:'
      a = 5.5d0
      b = 61.d0/11.d0
      do i=3,30
        c = b
        b = 111.d0 - 1130.d0/b + 3000.d0/(a*b)
        a = c
        write(*,'(a2,i2,a3,d23.16)')'U(',i,') = ',b
      enddo
      write(*,*)'The exact limit is 6.'
!
      end
